src/dbnode/storage/index/postings_list_cache.go

```diff
@@ -63,6 +63,14 @@ type PostingsListCache struct {
 	metrics *postingsListCacheMetrics
 }
 
+// PostingsListCacheQuery represents a query that we want to cache the result of
+// for a given segment. Note that it include the field in the segment that the
+// query was executed on, as well as the pattern of the query itself.
+type PostingsListCacheQuery struct {
```

Comment:
hm i think you need to include query/pattern type here. i.e. a term query of "abc:1*" is not the same as regexp query of "abc:1*". 

Could you ensure the tests catch this too.

Reply:
This is handled. They go through separate APIs and end up with separate keys in the LRU map. See postings_list_cache_lru.go key struct

Reply:
Fair enough. Mind including pattern here anyway? A query isn’t completely defined without it so this struct feels incomplete.

Reply:
I’d have to change the APIs though and then you have a potentially invalid iota coming from the caller

Reply:
Are you just opposing the name? We could call it TermPattern or FieldPattern

Reply:
Are you just opposing the name? We could call it TermPattern or FieldPattern

Reply:
It’s not the name as much as it is the struct itself. Like without the pattern included, the struct is an incomplete description of a query. 

How would you feel about migrating to using search.Query here? The thought is if we add new types of queries we’d still get this part for free.

Reply:
Search.Query is an interface :/ I don't think that makes a lot of sense.